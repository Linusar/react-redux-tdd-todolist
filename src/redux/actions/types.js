const START_APP = 'START_APP';
const ADD_ELEMENT = 'ADD_ELEMENT';
const REMOVE_ELEMENT = 'REMOVE_ELEMENT';

export { START_APP, ADD_ELEMENT, REMOVE_ELEMENT };
