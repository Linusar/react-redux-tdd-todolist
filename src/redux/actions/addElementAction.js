import { createAction } from 'redux-actions';
import { ADD_ELEMENT } from './types';

const addElementAction = createAction(ADD_ELEMENT, task => task);

export default addElementAction;
