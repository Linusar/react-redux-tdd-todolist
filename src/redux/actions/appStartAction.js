import { createAction } from 'redux-actions';
import { START_APP } from './types';

const appStartAction = createAction(START_APP, status => status);

export default appStartAction;
