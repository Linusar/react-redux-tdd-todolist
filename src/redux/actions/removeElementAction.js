import { createAction } from 'redux-actions';
import { REMOVE_ELEMENT } from './types';

const removeElementAction = createAction(REMOVE_ELEMENT, id => id);

export default removeElementAction;
