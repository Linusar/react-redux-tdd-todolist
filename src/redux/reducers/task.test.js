/* global describe expect test */
import Task from './task';
import { ADD_ELEMENT, REMOVE_ELEMENT } from '../actions/types';

describe('INITIAL_STATE', () => {
  test('is correct', () => {
    const action = { type: 'dummy_action' };

    expect(Task(undefined, action)).toMatchSnapshot();
  });
});

describe('ADD_ELEMENT', () => {
  test('returns the correct state', () => {
    const action = { type: ADD_ELEMENT, payload: 'First task' };

    expect(Task(undefined, action)).toMatchSnapshot();
  });
});

describe('REMOVE_ELEMENT', () => {
  test('returns the first element', () => {
    const action = { type: REMOVE_ELEMENT, payload: 12345 };

    expect(Task(undefined, action)).toMatchSnapshot();
  });
});
