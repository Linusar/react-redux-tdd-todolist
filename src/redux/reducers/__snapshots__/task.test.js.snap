// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ADD_ELEMENT returns the correct state 1`] = `
Array [
  Object {
    "key": 12345,
    "name": "First task",
  },
  "First task",
]
`;

exports[`INITIAL_STATE is correct 1`] = `
Array [
  Object {
    "key": 12345,
    "name": "First task",
  },
]
`;

exports[`REMOVE_ELEMENT returns the first element 1`] = `
Array [
  Object {
    "key": 12345,
    "name": "First task",
  },
]
`;
